src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.CinemaHallDao;
+import com.cinema.exception.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.CinemaHall;
+import com.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(cinemaHall);
+            transaction.commit();
+            return cinemaHall;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Unable to add " + cinemaHall + "to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<CinemaHall> allCinemaHallsQuery = session
+                    .createQuery("from cinema_halls ", CinemaHall.class);
```

Comment:
let's use class name, not the table name

Reply:
done